testing/src/test/java/io/stargate/it/http/docsapi/DocumentApiV2Test.java

```diff
@@ -54,46 +51,26 @@
   private String hostWithPort;
 
   @BeforeEach
-  public void setup(StargateConnectionInfo cluster) throws IOException {
+  public void setup(TestInfo testInfo, StargateConnectionInfo cluster) throws IOException {
     host = "http://" + cluster.seedAddress();
     hostWithPort = host + ":8082";
 
-    keyspace = "ks_docs_" + System.currentTimeMillis();
-    session =
-        CqlSession.builder()
-            .withConfigLoader(
-                DriverConfigLoader.programmaticBuilder()
-                    .withDuration(DefaultDriverOption.REQUEST_TRACE_INTERVAL, Duration.ofSeconds(5))
-                    .withDuration(DefaultDriverOption.REQUEST_TIMEOUT, Duration.ofSeconds(180))
-                    .withDuration(
-                        DefaultDriverOption.METADATA_SCHEMA_REQUEST_TIMEOUT,
-                        Duration.ofSeconds(180))
-                    .withDuration(
-                        DefaultDriverOption.CONNECTION_INIT_QUERY_TIMEOUT, Duration.ofSeconds(180))
-                    .withDuration(
-                        DefaultDriverOption.CONTROL_CONNECTION_TIMEOUT, Duration.ofSeconds(180))
-                    .build())
-            .withAuthCredentials("cassandra", "cassandra")
-            .addContactPoint(new InetSocketAddress(cluster.seedAddress(), 9043))
-            .withLocalDatacenter(cluster.datacenter())
-            .build();
-
-    assertThat(
-            session
-                .execute(
-                    String.format(
-                        "create keyspace if not exists %s WITH replication = "
-                            + "{'class': 'SimpleStrategy', 'replication_factor': 1 }",
-                        keyspace))
-                .wasApplied())
-        .isTrue();
+    Optional<String> name = testInfo.getTestMethod().map(Method::getName);
+    assertThat(name).isPresent();
+    String testName = name.get();
+    keyspace = "ks_docs_" + testName + "_" + System.currentTimeMillis();
+    // trim to max keyspace name length of 48
+    keyspace = keyspace.substring(0, Math.min(keyspace.length(), 48));
 
     initAuth();
-  }
 
-  @AfterEach
-  public void teardown() {
-    session.close();
+    String createKeyspaceRequest = String.format("{\"name\": \"%s\", \"replicas\": 1}", keyspace);
+
+    RestUtils.post(
+        authToken,
+        String.format("%s:8082/v2/schemas/keyspaces", host),
+        createKeyspaceRequest,
+        HttpStatus.SC_CREATED);
```

Comment:
I actually removed the use of the driver since it didn't seem to like being ran in parallel, there were alot of warnings about open sessions.

Reply:
That warning is just so that people don't forget to close their sessions in production applications. The default threshold (4) is intentionally aggressive so that issues are detected early.

 In our context, I think we can go up to a few dozen sessions in parallel without any issues. We can do that in `CqlSessionExtension.defaultConfig`:
```java
config.put(TypedDriverOption.SESSION_LEAK_THRESHOLD, 24);
```
It might take a bit of experimentation to find the sweet spot.